### ISSUE TITLE
[DataGridPro] Rows change the order when I move a column

### ISSUE THREAD

**Issue by GregRar** [mui/mui-x#4861]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When I use DataGrid PRO and when I drag a column header to change the order and move the cursor above the grid content rows it starts to reshuffle [attached video]. Our users are often working on pre-sorted data sets and it happens accidentally. How to disable it? This happens even if rowReordering is not enabled. Sorry, but I couldn't find an answer in the documentation.

### Expected behavior 🤔

Rows should not change the order.

### Steps to reproduce 🕹

https://user-images.githubusercontent.com/10992255/168254770-8d8ae565-61fc-4ef5-a305-9341fef379f7.mp4

https://deploy-preview-4034--material-ui-x.netlify.app/x/react-data-grid/rows/#customizing-the-row-reordering-icon

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by alexfauquette** [comment]
I think it has been fixed in #4735 which will be available in v5.11.0

You can test in the preview
https://deploy-preview-4735--material-ui-x.netlify.app/x/react-data-grid/rows/#customizing-the-row-reordering-icon

**Comment by GregRar** [comment]
Awesome, thank you Alexandre!
